util/include/Metrics.hpp

```diff
@@ -104,6 +107,40 @@ class Counter {
   uint64_t val_;
 };
 
+class AtomicCounter {
```

Comment:
Can we make the original classes such as `Counter` and `Gauge` templated and supply the variable type via the template?
Later we can do `using AtomicGauge = Gauge<std::atomic_uint64_t>;
WDYT?

Reply:
I tried, but it is too complicated and I'm not sure it worth the huge refactor we will have to do (also in concord)

Reply:
During an offline conversation, we've come up with another idea:
The original `Gauge` can be changed to:
```
template<typename CounterT>
class BasicGauge {
....
```
Then:
```
using Gauge = BasicGauge<uint64_t>;
using AtomicGauge = BasicGauge<std::atomic_uint64_t>;
```
As a result, all the old functionality using the original `Gauge` won't be affected.

Reply:
I like it. Great collaboration!

Reply:
done